### ISSUE TITLE
Cluster create fails the second and subsequent times 

### ISSUE THREAD

**Issue by bbergquist0930** [kubernetes-sigs/kind#3372]
**What happened**:

I issued 
```
kind create cluster
```
and the cluster is created.  I then issue 
```
kind delete cluster
kind create cluster
```
The cluster fails to create with error 
```
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.27.3) 🖼
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✗ Starting control-plane 🕹️ 
ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged kind-control-plane kubeadm init --skip-phases=preflight --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 1
```

What is strange is that if I reboot my laptop, the first `kind create cluster` will succees.  Subsequent ones will fail.

**What you expected to happen**:

The cluster work be created.  Note that I initially saw this while trying to create a cluster of 3 nodes and dropped back to just trying to create a cluster of the controller node to simplify the problem. 

**How to reproduce it (as _minimally_ and precisely as possible)**:

kind create cluster
kind delete cluster
kind create cluster

**Anything else we need to know?**:

**Environment:**

Dell Precision 7740 with 8 Core I9 processor

Memory:
```
              total        used        free      shared  buff/cache   available
Mem:       65591248     5140088    54249608     1205700     6201552    58509524
Swap:       8388604           0     8388604
```

Disk
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/nvme1n1p3  463G  191G  249G  44% /
brett@brett-Precision-7740:/tmp$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev             32G     0   32G   0% /dev
tmpfs           6.3G  6.4M  6.3G   1% /run
/dev/nvme1n1p3  463G  191G  249G  44% /
tmpfs            32G  216M   32G   1% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs            32G     0   32G   0% /sys/fs/cgroup
/dev/loop1      155M  155M     0 100% /snap/chromium/2636
/dev/loop2       56M   56M     0 100% /snap/core18/2790
/dev/loop3       64M   64M     0 100% /snap/core20/1974
/dev/loop4      155M  155M     0 100% /snap/chromium/2642
/dev/loop5       56M   56M     0 100% /snap/core18/2785
/dev/loop6       67M   67M     0 100% /snap/cups/980
/dev/nvme2n1    458G  1.5G  433G   1% /opt
/dev/nvme0n1    1.8T  305G  1.5T  18% /home
/dev/loop0      128K  128K     0 100% /snap/bare/5
/dev/loop7      165M  165M     0 100% /snap/gnome-3-28-1804/194
/dev/nvme1n1p1  826M   74M  753M   9% /boot/efi
/dev/loop9      165M  165M     0 100% /snap/gnome-3-28-1804/198
/dev/loop8       74M   74M     0 100% /snap/core22/858
/dev/loop10      67M   67M     0 100% /snap/cups/974
/dev/loop12     350M  350M     0 100% /snap/gnome-3-38-2004/140
/dev/loop13      64M   64M     0 100% /snap/core20/2015
/dev/loop11      74M   74M     0 100% /snap/core22/864
/dev/loop14      13M   13M     0 100% /snap/snap-store/959
/dev/loop15     219M  219M     0 100% /snap/gnome-3-34-1804/93
/dev/loop16      46M   46M     0 100% /snap/snap-store/638
/dev/loop17     350M  350M     0 100% /snap/gnome-3-38-2004/143
/dev/loop18      41M   41M     0 100% /snap/snapd/20092
/dev/loop19     425M  425M     0 100% /snap/kde-frameworks-5-qt-5-15-3-core20/8
/dev/loop20     497M  497M     0 100% /snap/gnome-42-2204/132
/dev/loop21     425M  425M     0 100% /snap/kde-frameworks-5-91-qt-5-15-3-core20/1
/dev/loop22      92M   92M     0 100% /snap/gtk-common-themes/1535
/dev/loop23      82M   82M     0 100% /snap/gtk-common-themes/1534
/dev/loop24     486M  486M     0 100% /snap/gnome-42-2204/126
/dev/loop25      41M   41M     0 100% /snap/snapd/19993
/dev/loop26     219M  219M     0 100% /snap/gnome-3-34-1804/90
tmpfs           6.3G   36K  6.3G   1% /run/user/1001
```

- kind version: (use `kind version`): kind v0.20.0 go1.20.4 linux/amd64
- Runtime info: (use `docker info` or `podman info`):
```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.11.2)
  compose: Docker Compose (Docker Inc., v2.21.0)
  scan: Docker Scan (Docker Inc., v0.23.0)

Server:
 Containers: 6
  Running: 2
  Paused: 0
  Stopped: 4
 Images: 23
 Server Version: 20.10.23
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 61f9fd88f79f081d64d6fa3bb1a0dc71ec870523
 runc version: v1.1.9-0-gccaecfc
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-163-generic
 Operating System: Ubuntu 20.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 62.55GiB
 Name: brett-Precision-7740
 ID: 4NVW:HGZJ:XX2D:C4FD:CDMQ:6OPJ:YIJD:72LF:LYOO:4K74:RZJI:BDL6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  docker-registry.everest.canoga.com
  192.168.49.0/24
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: API is accessible on http://0.0.0.0:2375 without encryption.
         Access to the remote API is equivalent to root access on the host. Refer
         to the 'Docker daemon attack surface' section in the documentation for
         more information: https://docs.docker.com/go/attack-surface/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```
- OS (e.g. from `/etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kubernetes version: (use `kubectl version`):
```
Client Version: v1.28.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```
- Any proxies or other special environment settings?:

**Comment by bbergquist0930** [comment]
[logs.zip](https://github.com/kubernetes-sigs/kind/files/12741407/logs.zip)

logs from running `kind export logs`

**Comment by mboutet** [comment]
I think I've experienced the same issue. Can you try enabling cgroup v2 on your machine? That fixed it for me.

Follow [these instructions](https://sleeplessbeastie.eu/2021/09/10/how-to-enable-control-group-v2/) starting at `To enable Control Group v2 alter boot options to append systemd.unified_cgroup_hierarchy=1 parameter.` and ignoring all the podman stuff.

**Comment by bbergquist0930** [comment]
Thanks @mboutet for your suggestion.  I actually had this option `systemd.legacy_systemd_cgroup_controller=true` in my boot options and I can't for the life of me remember why :disappointed: .  This sytem started as Ubuntu 18.04 and probably left over form there at some point with an issue with Docker maybe.

I did add the option that you mentioned but also removed the  `systemd.legacy_systemd_cgroup_controller=true` option at the same time.  And then I could create and delete the cluster over and over.  But I had changed two things at once, so to really know if it was the new option or removing the old option, I went back and just removed the `systemd.legacy_systemd_cgroup_controller=true` option.  I could still also create and delete the cluster over and over. To finely make sure that this was the reason, I re-added `systemd.legacy_systemd_cgroup_controller=true` option and I could no longer create a cluster after deleting it. 

So in my case it seems that the option  `systemd.legacy_systemd_cgroup_controller=true` is the problem.  I thank you because I would not have found this without your suggestion.

I have left in the `systemd.unified_cgroup_hierarchy=1` option as well with no ill effects.

Again, thank you!
